tests/TrackJS/TrackJSEmbedTest.php

```diff
@@ -0,0 +1,134 @@
+<?php declare(strict_types=1);
+
+use Becklyn\AssetsBundle\Helper\AssetHelper;
+use Becklyn\Hosting\Config\HostingConfig;
+use Becklyn\Hosting\Exception\AssetIntegrationFailedException;
+use Becklyn\Hosting\TrackJS\TrackJSEmbed;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Asset\Packages;
+
+class TrackJSEmbedTest extends TestCase
+{
+    public function testGetEmbedHtmlAssetsCheck () : void
+    {
+        /** @var MockObject|HostingConfig $hostingConfig */
+        $hostingConfig = $this->getMockBuilder(HostingConfig::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $trackJsEmbed = new TrackJSEmbed(
+            $hostingConfig,
+            null,
+            null,
+            false
+        );
+
+        $this->expectException(AssetIntegrationFailedException::class);
+
+        $trackJsEmbed->getEmbedHtml();
+    }
+
+    public function testGetEmbedHtmlInDebugMode () : void
```

Comment:
The test name doesn't really reflect what it's testing.

Reply:
Can you refactor the test to use a `dataProvider` to test the embed code in various scenarios?